### DISCUSSION TITLE
FluxCD recreates / deletes PVC if PVC size patched manually for deployments

### DISCUSSION THREAD

**Discussion by echel0n-HX** [fluxcd/flux2#4347] [General]
Not sure if this is the proper behavior of FluxCD v2 with PVCs but we just implemented a pvc-autoresizer that automatically patches the PVC size from the cluster side when a PVC only has 20% free space left, seems that when Flux reconciles from source it causes the PVC to be recreated.

Is there any way to prevent this from happening without ditching pvc autoresizing, we don't notice this issue when using statefulsets, seems only when using deployments / pvc's

Thanks!

**Comment by stefanprodan** [comment]
Flux undoes all in-cluster changes to fields defined in Git. I guess the deletion happens because you've set `force: true` in the Flux Kustomization or maybe that's how Kubernetes handles these changes by recreating the PVC. In any case, if the PVC is managed by a diffrent controller than Flux, you can set the annotation `kustomize.toolkit.fluxcd.io/ssa: IfNotPresent` on PVCs in Git. Flux will only create the PVC if it doesn't exists in cluster, and will ignore any changes going forward. Docs here: https://fluxcd.io/flux/components/kustomize/kustomizations/#controlling-the-apply-behavior-of-resources

Note that this only works for resources managed by Flux Kustomizations, If you use HelmReleases, Helm undoes all changes at upgrade.

**Comment by echel0n-HX** [comment]
Thats for all that and good to know, so your saying that manually increasing PVC size will be undone when using HelmReleases ?

**Reply by stefanprodan** [reply]
Yes, you can test this yourself, change something in the values.yaml then run `hem upgrade`. Flux does the same thing.

**Reply by stefanprodan** [reply]
Also `kubectl apply -f pvc.yaml` does the same stuff, it will revert the size to what's in the YAML, which is expected.

**Comment by echel0n-HX** [comment]
Thanks, I'll need to check further into this.

**Comment by echel0n-HX** [comment]
Last question, if switching from deployments to statefulsets, manually updating pvc size wouldn't/shouldn't be undone then correct?

**Reply by stefanprodan** [reply]
They should, if it doesn’t happen probably your stateful sets are blocking the PVCs changes.
